### PR TITLE
intel_adsp: ace: dmic power domain

### PR DESCRIPTION
**Pull request by tmleman**
Combining the Intel DMIC with proper power domain to prevent it power gaitin during d0i0/d0i3 transition.

* intel adsp power domain update,
* initial implementation for dmic runtime pm.

### CODE REVIEW

File: drivers/dai/intel/dmic/dmic.c [hunk 1]
```diff
@@ -811,9 +812,29 @@ static int dai_dmic_remove_wrapper(const struct device *dev)
...
+		break;
+	case PM_DEVICE_ACTION_RESUME:
+		dai_dmic_probe_wrapper(dev);
+		break;
+	case PM_DEVICE_ACTION_TURN_OFF:
+	case PM_DEVICE_ACTION_TURN_ON:
+		/* All device pm is handled during resume and suspend */
+		break;
+	default:
+		return -ENOTSUP;
```
Comment (marcinszkudlinski) [inline drivers/dai/intel/dmic/dmic.c:829]: why default is an error? What if another action type - not needed and not handled here, exactly as TURN_ON/OFF - is added sometimes?

Comment (tmleman) [reply drivers/dai/intel/dmic/dmic.c:849]: This is based on current implementation for a ssp. I believe that if action it's not supported it's better to learn about it by error when we are trying to trigger it and not in some other random place. If action should be supported we need add a implementation for it.

Comment (marcinszkudlinski) [reply drivers/dai/intel/dmic/dmic.c:849]: @nashif  what do you think about default action?

Comment (ceolin) [reply drivers/dai/intel/dmic/dmic.c:849]: I think returning `-ENOTSUP` is the right thing. This driver is currently handling all actions (even if it is not doing anything with `ON|OFF`) and
returning an error will let us know if a different state is added and may affect us. Most drivers do this btw.

**Review by ceolin — COMMENTED** [review]
Just to make it clear. `pm_device_init_suspended` is only needed if the device is *at this moment physically suspended.
After `pm_device_runtime_enable` the device will be **suspended** and we will need to call probe to resume it.

File: drivers/dai/intel/dmic/dmic.c [hunk 2]
```diff
@@ -852,7 +873,9 @@ static int dai_dmic_initialize_device(const struct device *dev)
 		dai_dmic_irq_handler,
 		DEVICE_DT_INST_GET(0),
 		0);
-	return 0;
+
+	pm_device_init_suspended(dev);
+	return pm_device_runtime_enable(dev);
```
Comment (ceolin) [inline drivers/dai/intel/dmic/dmic.c:878]: Just to make it clear. `pm_device_init_suspended` is only needed if the device is *at this moment physically suspended.
After `pm_device_runtime_enable` the device will be **suspended** and we will need to call probe to resume it.

**Review by ceolin — APPROVED** [review]
PM changes LGTM

**Review by dcpleung — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Pull request merged**